src/pages/vaults/VaultDepositWithdrawForm.tsx

```diff
@@ -138,16 +146,81 @@ export const VaultDepositWithdrawForm = ({
     dispatch(setVaultFormConfirmationStep(true));
   };
 
-  const onSubmitConfirmForm = () => {
-    // TODO tell abacus and respond
-    onSuccess?.();
+  const { depositToMegavault, withdrawFromMegavault } = useSubaccount();
+  const forceRefreshVaultAccount = useForceRefreshVaultAccount();
+  const forceRefreshVault = useForceRefreshVaultDetails();
+  const notify = useCustomNotification();
+
+  const onSubmitConfirmForm = async () => {
+    if (isSubmitting) {
+      return;
+    }
+    setIsSubmitting(true);
+    try {
+      const { submissionData } = validationResponse;
+      if (operation === 'DEPOSIT') {
+        const cachedAmount = submissionData?.deposit?.amount;
+        if (cachedAmount == null) {
+          notify({
+            title: 'Unable to submit Megavault Deposit transaction',
+            body: 'Please adjust the amount and try again. If the problem persists, try refreshing the page or contacting support.',
+          });
+          // eslint-disable-next-line no-console
+          console.error('Somehow got to submission with empty amount in validation response');
```

Comment:
send to datadog via `dd.log`

Reply:
in a follow up pr for error detection and monitoring more broadly